test/dlc_tests/ec2/test_efa.py

```diff
@@ -35,88 +36,125 @@
-    on pipelines.
-    :param pytorch_training: str PyTorch Training DLC image URI
-    :param efa_ec2_instances: list of tuples of instance-ids and SSH-keys for EFA-enabled instances
-    :param efa_ec2_connections: list of Fabric Connection objects for EFA-enabled instances
-    :param ec2_instance_type: str Instance Type being tested
-    :param region: str Region in which EFA-enabled instances are launched
-    :param gpu_only: pytest fixture to limit test only to GPU DLCs
-    """
-    number_of_nodes = 2
-    _setup_multinode_efa_instances(
-        pytorch_training, efa_ec2_instances, efa_ec2_connections, ec2_instance_type, region
-    )
-    master_connection = efa_ec2_connections[0]
-    run_cmd_on_container(MASTER_CONTAINER_NAME, master_connection, EFA_SANITY_TEST_CMD, hide=False)
+# @pytest.mark.processor("gpu")
+# @pytest.mark.model("N/A")
+# @pytest.mark.integration("efa")
+# @pytest.mark.usefixtures("sagemaker")
+# @pytest.mark.allow_p4de_use
+# @pytest.mark.multinode(2)
+# @pytest.mark.parametrize("ec2_instance_type,region", EC2_EFA_GPU_INSTANCE_TYPE_AND_REGION)
+# @pytest.mark.skipif(
+#     is_pr_context() and not is_efa_dedicated(),
+#     reason="Skip EFA test in PR context unless explicitly enabled",
+# )
+# def test_pytorch_efa(
+#     pytorch_training, efa_ec2_instances, efa_ec2_connections, ec2_instance_type, region, gpu_only
+# ):
+#     """
+#     Run EFA Sanity tests on DLC, and then run NCCL Message Transfer and All Reduce tests using EFA
+#     on multiple nodes using DLC images. The test scripts are agnostic to the framework and version
+#     installed in the DLC image. The test also builds nccl-tests to create the all_reduce_perf
+#     binary necessary for multinode tests, on each node.
+#     Note: This test must be explicitly enabled on CI, and will only run on EFA-capable instances
+#     on pipelines.
+#     :param pytorch_training: str PyTorch Training DLC image URI
+#     :param efa_ec2_instances: list of tuples of instance-ids and SSH-keys for EFA-enabled instances
+#     :param efa_ec2_connections: list of Fabric Connection objects for EFA-enabled instances
+#     :param ec2_instance_type: str Instance Type being tested
+#     :param region: str Region in which EFA-enabled instances are launched
+#     :param gpu_only: pytest fixture to limit test only to GPU DLCs
+#     """
+#     number_of_nodes = 2
+#     _setup_multinode_efa_instances(
+#         pytorch_training, efa_ec2_instances, efa_ec2_connections, ec2_instance_type, region
+#     )
+#     master_connection = efa_ec2_connections[0]
+#     run_cmd_on_container(MASTER_CONTAINER_NAME, master_connection, EFA_SANITY_TEST_CMD, hide=False)
+
+#     run_cmd_on_container(
+#         MASTER_CONTAINER_NAME,
+#         master_connection,
+#         f"{EFA_INTEGRATION_TEST_CMD} {HOSTS_FILE_LOCATION} {number_of_nodes}",
+#         hide=False,
+#         timeout=300,
+#     )
```

Comment:
Can you please uncomment all the commented test in test/dlc_tests/ec2/test_efa.py file in the upcoming commit?

Reply:
uncommented pytorch EFA test, pls note above test case has two run_cmd_on_container()